Implement revision diff

Komediruzecki opened the pull request:
Implement revision diff

Add diff npm lib
Update revision modal to check for diff between two revisions
Set CodeMirror mode to diff

Tested in electron and web app
- Restore revision
- Add new content, investigate revision between multiple versions

Fixes: https://github.com/Boostnote/BoostHub/issues/1321

Showcase:

https://user-images.githubusercontent.com/18196945/133827160-ce20dc87-1139-46b9-9462-d51cd31a1c2a.mp4

New showcase:
- Fixes error symbols, initial preview (deletions and additions)
- Background color for additions/deletions


https://user-images.githubusercontent.com/18196945/133894776-85012886-802c-4648-982d-6b87fe02727d.mp4

Rokt33r reviewed (changes requested):
<img width="996" alt="Screen Shot 2021-09-18 at 7 00 32 PM" src="https://user-images.githubusercontent.com/5865853/133885328-0bc32e9f-7b56-4bd6-b788-7667913b5261.png">

The existing `-` and `+` chars should be escaped when creating a diff patch.

If the syntax of the diff mode doesn't support escaping those chars, we should make a custom diff view component.

Or we could adopt this. 
https://codemirror.net/demo/merge.html#

Komediruzecki commented:
> <img alt="Screen Shot 2021-09-18 at 7 00 32 PM" width="996" src="https://user-images.githubusercontent.com/5865853/133885328-0bc32e9f-7b56-4bd6-b788-7667913b5261.png">
> 
> The existing `-` and `+` chars should be escaped when creating a diff patch.
> 
> If the syntax of the diff mode doesn't support escaping those chars, we should make a custom diff view component.
> 
> Or we could adopt this.
> https://codemirror.net/demo/merge.html#

Merge CodeMirror setup is complicated I think, and we don't really need it, I fixed the characters and updated the styling a bit - see showcase.

Rokt33r reviewed (approved):
LGTM!
